.github/workflows/toolchain.yml

```diff
@@ -0,0 +1,60 @@
+name: rust-toolchain
+
+on:
+  pull_request:
+    paths:
+      - rust-toolchain
+      - "**Dockerfile"
+      - ".github/workflows/*"
+      - ".github/**/Dockerfile"
+
+permissions:
+  contents: read
+
+env:
+  YQ_VERSION: v4.2.0
+
+jobs:
+  dockerfiles:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@ec3a7ce113134d7a93b817d10a8272cb61118579
+      - run: |
+          VERSION="$(cat rust-toolchain)"
+          ex=0
+          for f in $(find . -name Dockerfile -and -not -path './.github/actions/release-tag-meta/*'); do
+            versions=$(sed -nE 's/^ARG RUST_VERSION=([^ ]+)/\1/p' $f)
+            if [ -z "$versions" ]; then
+              echo "::error file=$f::$f missing 'RUST_VERSION' argument"
+              ex=$((ex + 1))
+            fi
+            for mismatch in $(echo "$version" | grep -vF "$VERSION" || true) ; do
+              echo "::error file=$f::$f uses incorrect rust version(s): $mismatch"
+              ex=$((ex + 1))
+            done
+          done
+          exit $ex
+
+  workflows:
+    runs-on: ubuntu-latest
+    steps:
```

Comment:
Might be a silly question but would the yq action simplify anything here?

https://github.com/mikefarah/yq/issues/844#issuecomment-856700574

I guess it might not be as easy to check the result if we used the action as opposed to doing it in bash during code checkout?